public/app/api/clients/folder/v1beta1/index.ts

```diff
@@ -1,8 +1,170 @@
+
+    const keys: string[] = result.data
+      ? [
+          result.data.metadata.annotations?.[AnnoKeyUpdatedBy],
+          result.data.metadata.annotations?.[AnnoKeyCreatedBy],
+        ].filter((v) => v !== undefined)
+      : [];
+
+    const resultUserDisplay = useGetDisplayMappingQuery(keys.length ? { key: keys } : skipToken);
+
+    let newData: FolderDTO | undefined = undefined;
+    if (result.data && resultParents.data && resultAccess.data && resultUserDisplay.data) {
+      const updatedBy = result.data.metadata.annotations?.[AnnoKeyUpdatedBy];
+      const createdBy = result.data.metadata.annotations?.[AnnoKeyCreatedBy];
+      newData = {
+        canAdmin: resultAccess.data.canAdmin,
+        canDelete: resultAccess.data.canDelete,
+        canEdit: resultAccess.data.canEdit,
+        canSave: resultAccess.data.canSave,
+        accessControl: resultAccess.data.accessControl,
+        created: result.data.metadata.creationTimestamp || '0001-01-01T00:00:00Z',
+        createdBy:
+          (createdBy && resultUserDisplay.data.display[resultUserDisplay.data.keys.indexOf(createdBy)]?.displayName) ||
+          'Anonymous',
+        // Does not seem like this is set to true in the legacy API
+        hasAcl: false,
+        id: parseInt(result.data.metadata.labels?.[DeprecatedInternalId] || '0', 10) || 0,
+        parentUid: result.data.metadata.annotations?.[AnnoKeyFolder],
+        managedBy: result.data.metadata.annotations?.[AnnoKeyManagerKind] as ManagerKind,
+
+        title: result.data.spec.title,
+        uid: result.data.metadata.name!,
+        updated: result.data.metadata.annotations?.[AnnoKeyUpdatedTimestamp] || '0001-01-01T00:00:00Z',
+        updatedBy:
+          (updatedBy && resultUserDisplay.data.display[resultUserDisplay.data.keys.indexOf(updatedBy)]?.displayName) ||
+          'Anonymous',
+        // Seems like this annotation is not populated
+        // url: result.data.metadata.annotations?.[AnnoKeyFolderUrl] || '',
+        // general folder does not come with url
+        // see https://github.com/grafana/grafana/blob/8a05378ef3ae5545c6f7429eae5c174d3c0edbfe/pkg/services/folder/folderimpl/folder_unifiedstorage.go#L88
+        url: uid === GENERAL_FOLDER_UID ? '' : folderUrl(result.data.metadata.name!, result.data.spec.title!),
+        version: result.data.metadata.generation || 1,
+      };
+
+      if (resultParents.data.items?.length) {
+        newData.parents = resultParents.data.items
+          .filter((i) => i.name !== result.data!.metadata.name)
+          .map((i) => ({
+            title: i.title,
+            uid: i.name,
+            // No idea how to make slug, on the server it uses a go lib: https://github.com/grafana/grafana/blob/aac66e91198004bc044754105e18bfff8fbfd383/pkg/infra/slugify/slugify.go#L56
+            // Don't think slug is needed for the URL to work though
+            url: folderUrl(i.name, i.title),
```

Comment:
We also have [this](https://github.com/grafana/grafana/blob/106206ae936e9ec64e575abfffeaf240a9b66abb/public/app/features/browse-dashboards/components/utils.ts#L15), possibly could be reused.

Reply:
We actually have multiple places where something similar is happening but usually without the slug for some reason. The slug probably isn't super important for the most part but it is needed for the tabs in `/dashboards` page to work
<img width="294" height="60" alt="Screenshot 2025-07-10 at 14 58 00" src="https://github.com/user-attachments/assets/0455f05d-b381-4134-96ca-0f5e8569fb79" />
As they have the slug as part of the route params.